cityofaustin/atd-moped
# Add GitHub workflow to export views to SQL based on the migrations in your branch

## Associated issues

This PR aims to close https://github.com/cityofaustin/atd-data-tech/issues/15704.

##  Process

I find the easiest way to develop GitHub workflows is to commit directly into `main`, because the state of the repo is a parameter to the functionality you're building. I also knew that while this was under development, I'd need to be creating a new feature branch via the workflow every time I tested it, so that I wouldn't have to re-de-format my SQL as the workflow overwrites the files and commits them back into the repo. I knew I didn't want to accidentally push up 40 useless branches by accident at all. 😅 So, I forked the repo into my personal account so I handle all those concerns. 

⚠️ All that to say, `frankhereford:main` is the feature branch I'm PRing into `cityofaustin:main`. 

## Testing

### Demonstration

See here: https://github.com/frankhereford/atd-moped/pull/4.

### Manual testing

This one is sort of different in terms of testing. It may be possible to read the code and convince yourself that it can't really harm anything, and let it be merged. The triggering event is pushing on a branch which is not `main` or `production`, and to really simulate that, you need to do the following. It's a mini pain:

* Fork [the upstream repo](https://github.com/frankhereford/atd-moped) in your own account
* Enable actions for the new repo
* Disable all other workflows except the one we're testing here because they need secrets your new repo doesn't/shouldn't have defined.
* Create a feature branch in the new repo and make a migration that changes a view
* When you push up the feature branch with the new migration, you should see your feature branch get commits from the "View Bot" and have the SQL files reflect your migrations.

## Question: Should I format the view SQL files as part of this PR?

I have formatted the existing view files with the workflow in here, so they will not generate unexpected changes for the first person who modifies a moped view after this makes it into `main`. Let me know if y'all'd like this handled differently - thanks!

---
#### Ship list
